backend/src/question/question.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from "@nestjs/common";
+import { Difficulty, Question } from "@prisma/client";
+
+import { PrismaService } from "src/core/prisma.service";
+
+import { DifficultyLevel } from "../../../shared/src/types/base";
+
+@Injectable()
+export class QuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  // gets a random question of diffiulty difficultyLevel
+  async getByDifficulty(
+    difficultyLevel: DifficultyLevel,
+  ): Promise<Question | null> {
```

Comment:
We can actually just return the ID here, but returning the whole question is also okay.